### PR TITLE
Support cljc

### PR DESCRIPTION
**Pull request by zerg000000**
* Dump Clojure version to 1.8.0 for cljc support
* Change all core source files suffix to .cljc
* Fix tests which base on wrong assumption

### CODE REVIEW

File: core/test/ragtime/core_test.cljc [hunk 1]
```diff
@@ -10,12 +10,12 @@
...
     (swap! data update-in [:migrations] conj id))
   (remove-migration-id [_ id]
-    (swap! data update-in [:migrations] disj id))
+    (swap! data update-in [:migrations] (fn [coll] (filterv #(not= % id) coll))))
   (applied-migration-ids [_]
     (seq (:migrations @data))))
 
 (defn in-memory-db []
-  (InMemoryDB. (atom {:migrations #{}})))
+  (InMemoryDB. (atom {:migrations []})))
```
Comment (weavejester) [inline core/test/ragtime/core_test.cljc]: Can you explain this change?

Comment (zerg000000) [reply core/test/ragtime/core_test.cljc]: The use of `set` which will not preserve the ordering of the migrations. However, functions using `(applied-migration-ids db)` are assuming the ids returned in applied order.

For example, `(rollback-to store index x-id)` rely on the correct ordering of the applied migrations, it simply reverse the applied migration ids, `(down)` it one by one until x-id is found.

Therefore I change `set` to `vector` to keep the ordering.

Comment (weavejester) [reply core/test/ragtime/core_test.cljc]: Okay, thanks - that does look like a mistake. However, can you put that fix in another PR? It doesn't seem to be directly related to supporting cljc, but instead seems like a general fix.

**Comment by weavejester** [comment]
Can you also explain your use-case for needing cljc support?

**Comment by zerg000000** [comment]
> Can you also explain your use-case for needing cljc support?

I am experimenting to implement a cljs version of `ragtime` based on node library `pg`. But find the non-blocking pg API is incompatible with `ragtime`, so support .cljc is not a valid option anymore. I will extract the unit test fix in a separate PR. Thanks for your time.

**Comment by zerg000000** [comment]
Create #138 #139 for the minor fixes. Close this issue, since no use-case for supporting cljc.

**Pull request closed without merging**